js/kafka/kafka-client.js

```diff
@@ -38,6 +38,17 @@ export class KafkaClient extends Client {
         return topic.replace(illegalChars, "-");
     }
 
+    /**
+     * Decomposes a topic name to extract it's global topic name.
+     *
+     * @param {String} topic The name of the topic to be decomposed.
+     * @returns {String} The global topic name.
+     */
+    static getGlobalTopic(topic) {

```

Comment:
I think this is actually quite generic logic for all adapters since this is only going from topic name to global topic, I would move this to the base.js